### PR TITLE
[12.0][MIG] Migrate base_mixin_restrict_field_access from 8.0

### PR DESCRIPTION
**Pull request by RobinetDenisAcsone**
Waiting PR:
- [x] base_suspend_security  (https://github.com/OCA/server-backend/pull/31)

### CODE REVIEW

File: base_mixin_restrict_field_access/models/restrict_field_access_mixin.py [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+                domain
+            )
+
+        return super(RestrictFieldAccessMixin, self).read_group(
+            domain, fields, groupby, offset=offset, limit=limit,
+            orderby=orderby, lazy=lazy
+        )
+
+    @api.multi
+    def _export_rows(self, fields, *, _is_toplevel_call=True):
```
Comment (RobinetDenisAcsone) [inline base_mixin_restrict_field_access/models/restrict_field_access_mixin.py:124]: It was `_BaseModel__export_rows`, I don't know what's the difference between this call and the the direct call.
Note (don't know if it's related): in v8 it was `__export_rows` (2 _), and not `_export_rows`

cc @hbrunn

Comment (hbrunn) [reply base_mixin_restrict_field_access/models/restrict_field_access_mixin.py:124]: that was some inheritance-foo in v8, if this works just like that in v12 then we're happy

**Comment by dannyadair** [comment]
> Hey @RobinetDenisAcsone, thank you for your Pull Request.
> 
> It looks like some users haven't signed our **C**ontributor **L**icense **A**greement, yet.
> You can read and sign our full Contributor License Agreement here: http://odoo-community.org/page/cla
> Here is a list of the users:
> 
> * @dannyadair (login unknown in OCA database)
> 
> Appreciation of efforts,
> OCA CLAbot

Sent signed ICLA and ECLA

File: base_mixin_restrict_field_access/models/restrict_field_access_mixin.py [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+        result = super(RestrictFieldAccessMixin, self).read(
+            fields=fields, load=load)
+        for record in result:
+            this = self.browse(record['id'])
+            for field in record:
+                if not this._restrict_field_access_is_field_accessible(field):
+                    record[field] = self._fields[field].convert_to_read(
+                        self._fields[field].null(self), record)
+                    if self.env.cache.get(self, self._fields[field]):
+                        self.env.cache.remove(self, self._fields[field])
```
Comment (grindtildeath) [inline base_mixin_restrict_field_access/models/restrict_field_access_mixin.py:77]:
```suggestion
                    if self.env.cache.get(this, self._fields[field]):
                        self.env.cache.remove(this, self._fields[field])
```

**Review by grindtildeath — CHANGES_REQUESTED** [review]
Thanks for the migration.
However, it seems to need some fixing and the test must be improved as it didn't catch the singleton exception on read.
@OCA/tools-maintainers Can someone tag this PR as Needs fixing please?

**Pull request closed without merging**